docs/migrate-to-ory/auth0.mdx

```diff
@@ -0,0 +1,263 @@
+        fi
+        poll_job_status
+
+    elif [[ $state == "completed" ]]; then
+        location=$(echo $jobstatus | jq -r ".location")
+        curl "$location" --silent --output "AUTH0_USERDATA_nd.json.gz"
+        gzip -d -c "AUTH0_USERDATA_nd.json.gz" | jq -s "." >"AUTH0_USERDATA.json"
+        echo "Finished downloading Auth0 user data!"
+    fi
+}
+poll_job_status
+```
+
+## Creating Identities
+
+Once the download is finished and Auth0 provided you with the hashed passwords, we can create the identities
+`in Ory Kratos / Ory Network?`.
+
+We'll create the identities using an identity schema that's requires the users to login via email and password.
+
+Since it's possible your user data contains new users who weren't registered when the hashed passwords were exported, we'll
+reserve an identity for that email. If you're on a free Auth0 plan and you don't qualify for an export of your hashed passwords,
+then this might be a viable option for you (`pw reset flow?`).
+
+Now we create an Ory Project incase you don't have one yet:
+
+`setting schema doesnt work this way and needs to be updated`
+
+```bash
+ory create project --name "Ory Docs Auth0 Migration Example"
+project_id="{set to the project ID from output}"
+
+# Use the email identity schema preset
+ory patch identity-config $project_id \
+  --replace '/identity/default_schema_id="preset://email"' \
+  --replace '/identity/schemas=[{"id":"preset://email","url":"preset://email"}]'
+```
+
+We export the project slug and create a PAT `how?` and add paths pointing to the Auth0 exports.
+
+```bash
+export ORY_SLUG='{your_ory_slug}'
+export ORY_TOKEN='{ory_token}'
+export AUTH0_USERDATA="{/path/to/AUTH0_USERDATA.json}"
+export AUTH0_PWEXPORT="{/path/to/hashed_passwords.json}" # comment this line to reserve not create identities
+```
+
+Now we can create the identities using this script.
+
+```bash
+create_payload() {
+    unset payload
+
+    # uses both email and pwhash. pwhash can also be 'null'-string
+    if [[ -z "$email" && -z "$pwhash" ]]; then
+        echo "please supply both an email and password"
+        exit 1
+    elif [[ -z "$pwhash" || "$pwhash" == "null" ]]; then
+        payload=$(jq -n \
+            --arg sid "$ORY_SCHEMA_ID" \
+            --arg em "$email" \
+            '{schema_id: $sid, traits: {email: $em}}')
+    else
+        payload=$(
+            jq -n \
+                --arg sid "$ORY_SCHEMA_ID" \
+                --arg em "$email" \
+                --arg pwhash "$pwhash" \
+                '{schema_id: $sid,
+                  traits:
+                    {email: $em},
+                     credentials:
+                         {password:
+                             {config:
+                                 {hashed_password: $pwhash}}}}'
+        )
+    fi
+}
+
+create_identity() {
+    if [[ -z "$payload" ]]; then
+        echo "please supply a valid payload"
+        exit 1
+    else
+        curl --request POST -sL \
```

Comment:
Consider using `ory import identities` instead, it allows importing many identities from a JSON source